mobile/native/src/channel_fee.rs

```diff
@@ -110,30 +123,64 @@ impl ChannelFeePaymentSubscriber {
             Handle::current().block_on(fetch_funding_transaction(txid))
         })?;
         tracing::debug!("Successfully fetched transaction fees of {} for new inbound channel with id {channel_id_as_str}", transaction.fee);
-        self.set_funding_transaction(transaction);
+        self.set_open_channel_info(channel_id, transaction);
         Ok(())
     }
 
-    fn set_funding_transaction(&self, transaction: EsploraTransaction) {
+    fn set_open_channel_info(&self, channel_id: &ChannelId, transaction: EsploraTransaction) {
         *self
-            .open_channel_tx
+            .open_channel_info
             .lock()
-            .expect("Mutex to not be poisoned") = Some(transaction);
+            .expect("Mutex to not be poisoned") = Some((*channel_id, transaction));
     }
 
-    fn unset_funding_transaction(&self) {
+    fn unset_open_channel_info(&self) {
         *self
-            .open_channel_tx
+            .open_channel_info
             .lock()
             .expect("Mutex to not be poisoned") = None;
     }
 
-    fn get_funding_transaction(&self) -> Option<EsploraTransaction> {
-        self.open_channel_tx
+    fn get_open_channel_info(&self) -> Option<(ChannelId, EsploraTransaction)> {
+        self.open_channel_info
             .lock()
             .expect("Mutex to not be poisoned")
             .clone()
     }
+
+    async fn wait_for_outbound_capacity(
+        &self,
+        channel_id: ChannelId,
+        funding_tx_fees_msats: u64,
+    ) -> Result<()> {
+        tokio::time::timeout(Duration::from_secs(5), async {
+            loop {
+                let channel_details = match self
+                    .channel_manager
+                    .get_channel_details(&channel_id) {
+                    Some(channel_details) => channel_details,
+                    None => {
+                        bail!("Could not find channel details for {}", hex::encode(channel_id));
+                    },
+                };
+
+                if channel_details.outbound_capacity_msat >= funding_tx_fees_msats {
```

Comment:
Do we need to check if the channel is `usable` aswell?

Reply:
The channel was always `usable`..